File: Seccion 12 - Punteros en C++/Seccion_12_Python/09_ejercicio_09.py

nFilas: int = 0
nCols: int = 0
matriz1: list = []
matriz2: list = []

def pedir_datos():
    global matriz1, matriz2, nFilas, nCols
    
    nFilas = int(input('Digite el número de filas: '))
    nCols = int(input('Digite el número de columnas: '))
    
    print('\nDigite los elementos de la primera matriz:\n')
    for i in range(nFilas):
        fila: list = []
        for j in range(nCols):
            val = int(input('Digite un número[{}][{}]: '.format(i, j)))
            fila.append(val)
        matriz1.append(fila)
    
    print('\nDigite los elementos de la segunda matriz:\n')
    for i in range(nFilas):
        fila: list = []
        for j in range(nCols):
            val = int(input('Digite un número[{}][{}]: '.format(i, j)))
            fila.append(val)
        matriz2.append(fila)

def sumar_matrices():
    global matriz1
    
    for i in range(nFilas):
        for j in range(nCols):
            matriz1[i][j] += matriz2[i][j]

def mostrar_matriz():
    print("\nLa matriz resultante es:\n")
    for i in range(nFilas):
        for j in range(nCols):
            print("{} ".format(matriz1[i][j]),end='')
        print()

def main():
    
    pedir_datos()
    
    sumar_matrices()
    
    mostrar_matriz()

if __name__ == '__main__':
    main()

File: Seccion 12 - Punteros en C++/Seccion_12_Dart/09_ejercicio_09.dart

import 'dart:io';

int nFilas, nCols;
List<List> matriz1 = new List<List>();
List<List> matriz2 = new List<List>();

void pedirDatos() {
  List<int> fila = new List<int>();

  stdout.write('Digite el número de filas: ');
  nFilas = int.parse(stdin.readLineSync());
  stdout.write('Digite el número de columnas: ');
  nCols = int.parse(stdin.readLineSync());

  print('\nDigite los elementos de la primera matriz:\n');
  for (int i = 0; i < nFilas; i++) {
    for (int j = 0; j < nCols; j++) {
      stdout.write('Digite un número[$i][$j]: ');
      fila.insert(j, int.parse(stdin.readLineSync()));
    }
    matriz1.insert(i, new List<int>.from(fila));
    fila.clear();
  }

  print('\nDigite los elementos de la segunda matriz:\n');
  for (int i = 0; i < nFilas; i++) {
    for (int j = 0; j < nCols; j++) {
      stdout.write('Digite un número[$i][$j]: ');
      fila.insert(j, int.parse(stdin.readLineSync()));
    }
    matriz2.insert(i, new List<int>.from(fila));
    fila.clear();
  }
}

void sumaMatrices(List<List> m1, List<List> m2, int nFilas, int nCols) {
  for (int i = 0; i < nFilas; i++) {
    for (int j = 0; j < nCols; j++) {
      m1[i][j] += m2[i][j];
    }
  }
}

void mostrarMatriz(List<List> m, int nFilas, int nCols) {
  print('\nLa matriz resultante es:\n');
  for (int i = 0; i < nFilas; i++) {
    for (int j = 0; j < nCols; j++) {
      stdout.write('${m[i][j]} ');
    }
    print('');
  }
}

void main() {

  pedirDatos();

  sumaMatrices(matriz1, matriz2, nFilas, nCols);

  mostrarMatriz(matriz1, nFilas, nCols);

}